test_integration/integration_tests.js

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+/**
+ * @file Truffle's `artifacts.require()` won't be available inside the Mocha
+ * tests. Thus, we load them here and add them to the `shared` object.
+ */
+const Mocha = require('mocha');
+const shared = require('./shared');
+
+const Anchor = artifacts.require('Anchor');
+const EIP20Gateway = artifacts.require('EIP20Gateway');
+const EIP20CoGateway = artifacts.require('EIP20CoGateway');
+const EIP20StandardToken = artifacts.require('EIP20StandardToken');
+
+const setupArtifacts = () => {
+    shared.artifacts = {
+        Anchor,
+        EIP20Gateway,
+        EIP20CoGateway,
+        EIP20StandardToken,
```

Comment:
We have more contracts that will be needed for testing. Why `OSTPrime`, `MessageBus`,`GatewayLib` ... etc are not included in this?

Reply:
This is about integration tests. So what we want to do is test the behavior of the system from the consumer's point of view. The consumer interacts with the anchor (anchor state root), the gateways (stake and redeem) and an EIP20StandardToken (approve transfer).
The user would not interact with the message bus or the gateway lib. They are, however, used by the others and should produce correct results, which will be asserted.

About OSTPrime I am not sure. Do we make calls to it directly? E.g. to assert balances? It is right now not part of the deployment process. Should it be added?

Reply:
OK 👍 Can we skip it here, then @deepesh-kn? Will be added in @sarvesh-ost's PR.

Reply:
👍